### PR TITLE
Feature/tfn 379 select journey selection

### PR DESCRIPTION
**Pull request by tejpowar**
# Description

For the return single service we require an inbound and outbound journey selection which replaces the direction page for this part of the journey.

As part of this pr i have created a directionDropdown component which we can reuse.

https://infinityworks.atlassian.net/browse/TFN-379

# Testing instructions

- Start the journey 
- On the fareType page selected "Return - Single Service"
- Service page make a selection
- Should redirect to the new page which is the Select Journey Direction
- Testing for errors - such as selecting one dropdown and not another
- As there is no matching page for outbound and inbound yet the journey will not be completed

Lighthouse test
Screenshot 2020-04-27 at 10 56 20

# Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

# Checklist:

-   [x] Able to run pr locally
-   [x] Lighthouse performed
-   [x] Followed acceptance criteria
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have made corresponding changes to the documentation if applicable
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/selectJourneyDirection.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                        className="govuk-button govuk-button--start"
+                    />
+                </form>
+            </main>
+        </Layout>
+    );
+};
+
+export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
+    deleteCookieOnServerSide(ctx, JOURNEY_COOKIE);
```
Comment (RobCatton) [inline src/pages/selectJourneyDirection.tsx:91]: Do we need to delete this on the server side? Other pages aren't doing this, generally, so I'm unfamiliar with why we would do it.

File: src/pages/api/selectJourneyDirection.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export default (req: NextApiRequest, res: NextApiResponse): void => {
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('Session is invalid.');
+        }
+
+        const { inboundJourney, outboundJourney } = req.body;
+
+        if (inboundJourney && outboundJourney) {
```
Comment (RobCatton) [inline src/pages/api/selectJourneyDirection.ts:15]: Do we want to do anything to check if what they have said the inboundJourney and outboundJourney is valid? Currently this API will accept any values, unsure if we want to throw an error if its not one of the valid choices for what they could select. We could reuse the `const rawService: RawService = await getServiceByNocCodeAndLineName(operatorInfo.nocCode, lineName);
    const service: Service = {
        ...rawService,
        journeyPatterns: await enrichJourneyPatternsWithNaptanInfo(rawService.journeyPatterns),
    };` methods from the props?

File: src/pages/selectJourneyDirection.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    return (
+        <Layout title={title} description={description}>
+            <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
+                <form action="/api/selectJourneyDirection" method="post">
+                    <ErrorSummary errors={errors} />
+                    <div className={`govuk-form-group ${errors.length > 0 ? 'govuk-form-group--error' : ''}`}>
+                        <fieldset className="govuk-fieldset" aria-describedby="page-heading">
+                            <legend className="govuk-fieldset__legend govuk-fieldset__legend--xl">
+                                <h1 className="govuk-fieldset__heading" id="page-heading">
+                                    Please select your journey direction
```
Comment (RobCatton) [inline src/pages/selectJourneyDirection.tsx:37]: Need to remove the word Please, no politeness allowed!

File: src/pages/api/selectJourneyDirection.ts (hunk 2 above)
Comment (tejpowar) [reply src/pages/api/selectJourneyDirection.ts:15]: as per conversation we wont do this for now as not beneficial due to the cookies which could get changed

File: src/pages/selectJourneyDirection.tsx (hunk 1 above)
Comment (tejpowar) [reply src/pages/selectJourneyDirection.tsx:91]: this was taken from direction page not sure why it is being done there also. I have removed it from mine

**Pull request merged**